pkg/engine/base_structures.go

```diff
@@ -77,3 +77,12 @@ func (resp *BaseResponse) Err() error {
 type SyncResponse struct {
 	BaseResponse
 }
+
+func (resp *SyncResponse) canRemoveGame() bool {
+	err := resp.Err()
+	if err == nil {
+		return false
+	}
+	_, panicOccured := err.(*ExecutionPanicError)
+	return panicOccured
+}
```

Comment:
Can we only remove games that have errors? I'm not 100% sure how this works

Reply:
See the interface definition:
```go
// Responses implementing this interface may indicate to the sender
// that the the game with `GameID()` should be removed from the engine.
type ResponseGameRemovable interface {
    canRemoveGame() bool
}
```

Basically, `canRemoveGame()` is a way to inform the engine that there is no point in the user making any more requests on that game.
For `PlayTurnResponse` that is the case when there are no more tiles to place as that indicates that the game is finished.
With the changes introduced in this PR, this is now also the case when a panic occurs during execution since it can leave the game in an inconsistent state.